sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,228 @@
+
+    #When axes are different and parent joint is at masscenter but child joint is at a unit vector from
+    #child masscenter.
+    N, A, P, C = generate_body()
+    PinJoint('J', P, C, child_axis=A.y, child_joint_pos=A.x)
+
+    assert A.y.angle_between(N.x) == 0 #Axis are aligned
+    assert A.y.express(N) == N.x
+    assert A.dcm(N) == Matrix([[0, -cos(theta), -sin(theta)], [1, 0, 0], [0, -sin(theta), cos(theta)]])
+    assert A.ang_vel_in(N) == omega*N.x
+    assert A.ang_vel_in(N).express(A) == omega * A.y
+    assert A.ang_vel_in(N).magnitude() == Abs(omega)
+    assert expand_mul(A.ang_vel_in(N).angle_between(A.y)) == acos(omega/Abs(omega))
+    assert C.masscenter.vel(N) == omega*A.z
+    assert C.masscenter.pos_from(P.masscenter) == -A.x
+    assert C.masscenter.pos_from(P.masscenter).express(N).simplify() == cos(theta)*N.y + sin(theta)*N.z
+    assert C.masscenter.vel(N).angle_between(A.x) == pi/2
+
+    #Similar to previous case but wrt parent body
+    N, A, P, C = generate_body()
+    PinJoint('J', P, C, parent_axis=N.y, parent_joint_pos=N.x)
+
+    assert N.y.angle_between(A.x) == 0 #Axis are aligned
+    assert N.y.express(A) ==  A.x
+    assert A.dcm(N) == Matrix([[0, 1, 0], [-cos(theta), 0, sin(theta)], [sin(theta), 0, cos(theta)]])
+    assert A.ang_vel_in(N) == omega*N.y
+    assert A.ang_vel_in(N).express(A) == omega* A.x
+    assert A.ang_vel_in(N).magnitude() == Abs(omega)
+    assert expand_mul(A.ang_vel_in(N).angle_between(A.x)) == acos(omega/Abs(omega))
+    assert C.masscenter.vel(N).simplify() == - omega*N.z
+    assert C.masscenter.pos_from(P.masscenter) == N.x
+
+    #Both joint pos id defined but different axes
+    N, A, P, C = generate_body()
+    PinJoint('J', P, C, parent_joint_pos=N.x, child_joint_pos=A.x, child_axis=A.x+A.y)
+    assert expand_mul(N.x.angle_between(A.x + A.y)) == 0 #Axis are aligned
+    assert (A.x + A.y).express(N).simplify() == sqrt(2)*N.x
+    assert A.dcm(N).simplify() == Matrix([[sqrt(2)/2, -sqrt(2)*cos(theta)/2, -sqrt(2)*sin(theta)/2], \
+        [sqrt(2)/2, sqrt(2)*cos(theta)/2, sqrt(2)*sin(theta)/2], [0, -sin(theta), cos(theta)]])
+    assert A.ang_vel_in(N) == omega*N.x
+    assert A.ang_vel_in(N).express(A).simplify() == (omega*A.x + omega*A.y)/sqrt(2)
+    assert A.ang_vel_in(N).magnitude() == Abs(omega)
+    assert expand_mul(A.ang_vel_in(N).angle_between(A.x + A.y)) == acos(omega/Abs(omega))
+    assert C.masscenter.vel(N).simplify() == (omega * A.z)/sqrt(2)
+    assert C.masscenter.pos_from(P.masscenter) == N.x - A.x
+    assert C.masscenter.pos_from(P.masscenter).express(N).simplify() == \
+        (1 - sqrt(2)/2)*N.x + sqrt(2)*cos(theta)/2*N.y + sqrt(2)*sin(theta)/2*N.z
+    assert C.masscenter.vel(N).express(N).simplify() == - sqrt(2)*omega*sin(theta)/2*N.y + sqrt(2)*omega*cos(theta)/2*N.z
+    assert C.masscenter.vel(N).angle_between(A.x) == pi/2
+
+    N, A, P, C = generate_body()
+    PinJoint('J', P, C, parent_joint_pos=N.x, child_joint_pos=A.x, child_axis=A.x+A.y-A.z)
```

Comment:
Here is my figure for this orientation:
![image](https://user-images.githubusercontent.com/276007/123770033-32128500-d8ca-11eb-8f78-fa22713f82f3.png)

Reply:
Here's my code for this one:
```
In [1]: import sympy as sm

In [2]: import sympy.physics.mechanics as me

In [3]: N = me.ReferenceFrame('N')

In [4]: A = me.ReferenceFrame('A')

In [5]: As = me.ReferenceFrame('As')

In [6]: a1, a2, a3 = sm.symbols('a1, a2, a3')

In [14]: rC = a1*N.x + a2*N.y + a3*N.z

In [15]: uC = me.cross(N.x, rC).normalize()

In [16]: thetaC = sm.acos(me.dot(N.x, rC.normalize()))

In [17]: As.orient_axis(N, -uC, thetaC)

In [19]: sm.trigsimp(As.dcm(N)).xreplace({a1: 1, a2: 1, a3: -1})
Out[19]:
Matrix([
[ sqrt(3)/3,      -sqrt(3)/3,       sqrt(3)/3],
[ sqrt(3)/3, sqrt(3)/6 + 1/2, 1/2 - sqrt(3)/6],
[-sqrt(3)/3, 1/2 - sqrt(3)/6, sqrt(3)/6 + 1/2]])

In [21]: theta = me.dynamicsymbols('theta')

In [22]: A.orient_axis(As, N.x, theta)

In [24]: sm.trigsimp(A.dcm(N).xreplace({a1: 1, a2: 1, a3: -1}))
Out[24]:
Matrix([
[ sqrt(3)/3,   -sqrt(6)*sin(theta(t) + pi/4)/3,    sqrt(6)*cos(theta(t) + pi/4)/3],
[ sqrt(3)/3,   sqrt(6)*cos(theta(t) + pi/12)/3,   sqrt(6)*sin(theta(t) + pi/12)/3],
[-sqrt(3)/3, sqrt(6)*cos(theta(t) + 5*pi/12)/3, sqrt(6)*sin(theta(t) + 5*pi/12)/3]])

In [26]: A.ang_vel_in(N).express(N).simplify()
Out[26]: Derivative(theta(t), t)*N.x

In [28]: A.ang_vel_in(N).express(A).simplify().xreplace({a1: 1, a2: 1, a3: -1})
Out[28]: sqrt(3)*Derivative(theta(t), t)/3*A.x + sqrt(3)*Derivative(theta(t), t)/3*A.y - sqrt(3)*Derivative(theta(t), t)/3*A.z
```

Reply:
This matches your results.